src/lib.rs

```diff
@@ -47,52 +47,171 @@
 #![forbid(non_ascii_idents, unsafe_code)]
 #![warn(clippy::nonstandard_macro_braces)]
 
+// Not public, but exported API. For macro expansion internals only.
+#[doc(hidden)]
+pub mod __private {
+    #[cfg(feature = "debug")]
+    pub use crate::fmt::{debug_tuple, DebugTuple};
+
+    #[cfg(feature = "error")]
+    pub use crate::vendor::thiserror::aserror::AsDynError;
+}
+
+/// Module containing macro definitions only, without corresponding traits.
+///
+/// Use it in your import paths, if you don't want to import traits, but only macros.
+pub mod macros {
+    #[doc(inline)]
+    pub use derive_more_impl::*;
+}
+
+// The modules containing error types and other helpers
+#[cfg(any(feature = "add", feature = "not"))]
+pub mod ops;
+
+#[cfg(feature = "debug")]
+mod fmt;
+
+#[cfg(feature = "error")]
+mod vendor;
+
+#[cfg(feature = "from_str")]
+mod r#str;
+#[cfg(feature = "from_str")]
 #[doc(inline)]
-pub use derive_more_impl::*;
+pub use crate::r#str::FromStrError;
 
 #[cfg(feature = "try_into")]
 mod convert;
 #[cfg(feature = "try_into")]
-pub use self::convert::TryIntoError;
+#[doc(inline)]
+pub use crate::convert::TryIntoError;
 
 #[cfg(feature = "try_unwrap")]
 mod try_unwrap;
 #[cfg(feature = "try_unwrap")]
+#[doc(inline)]
 pub use self::try_unwrap::TryUnwrapError;
 
-#[cfg(feature = "debug")]
-mod fmt;
 
-#[cfg(any(feature = "add", feature = "not"))]
-pub mod ops;
+// When re-exporting traits from std we need to do a pretty crazy trick, because we ONLY want
+// to re-export the traits and not derives that are called the same in the std module,
+// because those would conflict with our own. The way we do this is by first importing both
+// the trait and possible derive into a separate module and re-export them. Then we wildcard import
+// all the things from that module into the main module, but we also import our own derive by its
+// exact name. Due to the way wildcard imports work in rust, that results in our own derive taking
+// precedence over any derive from std.
+macro_rules! re_export_traits((
+    $feature:literal, $new_module_name:ident, $module:path $(, $traits:ident)* $(,)?) => {
+        #[cfg(feature = $feature)]
+        mod $new_module_name {
+            pub use $module::{$($traits),*};
+        }
 
-#[cfg(feature = "from_str")]
-mod r#str;
-#[cfg(feature = "from_str")]
-pub use self::r#str::FromStrError;
+        #[cfg(feature = $feature)]
+        #[doc(hidden)]
```

Comment:
I really disliked how the standard library trait listing was taking up a third of the front page of the documentation. Especially beacuse they came before the documentation of the actual derives. So I hid them again and I added a small snippet to the README to say that we re-export all traits.